subsys/bluetooth/scan.c

```diff
@@ -791,6 +814,104 @@ static int scan_appearance_filter_add(u16_t appearance)
 	return 0;
 }
 
+static bool adv_manufacturer_data_cmp(const u8_t *data,
+		u8_t data_len,
+		const u8_t *target_data,
+		u8_t target_data_len)
+{
+	if (target_data_len > data_len) {
```

Comment:
`target_data_len` can be shorter than `data_len` ?

Reply:
Yes, see the updated commit message. If you want to scan for, e.g., an iBeacon, you only need the UUID of the advertisement package, but not all the other information like major or minor number etc. which is also in the manufacturer data package.